pkg/cri/server/sandbox_run.go

```diff
@@ -281,6 +267,39 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		return nil, fmt.Errorf("failed to get sandbox container info: %w", err)
 	}
 
+	if podNetwork {
+		netStart := time.Now()
+		defer func() {
+			if retErr != nil {
+				deferCtx, deferCancel := ctrdutil.DeferContext()
+				defer deferCancel()
+				// Teardown network if an error is returned.
+				if err := c.teardownPodNetwork(deferCtx, sandbox); err != nil {
+					log.G(ctx).WithError(err).Errorf("Failed to destroy network for sandbox %q", id)
+					shouldDelContainer = false
+				}
+
+				if err := sandbox.NetNS.Remove(); err != nil {
```

Comment:
Good catch, I should have made this defer return if teardownPodNetwork fails.

Reply:
After double check, even if the network namespace is cleaned, the `teardownPodNetwork` will still succeed in sandbox_stop.go. Code ref: https://github.com/containerd/containerd/blob/v1.6.4/pkg/cri/server/sandbox_stop.go#L85

If network namespace is cleaned, `sandbox.NetNS.Closed()` will be true. `c.teardownPodNetwork` will return success.

if network namespace is not cleaned, `sandbox.NetNS.Closed()` will be false.  `c.teardownPodNetwork` will return success too.

Do you have any preference?

Reply:
It is specific to CNI implementation IIRC. We should keep the network namespace so that CNI plugin that relies on it to do proper cleanup can still do it.

Please note that even just based on the code, `teardownPodNetwork` does use the ns path. https://github.com/containerd/containerd/blob/902212651b12e4d81b7bc779632b2e4da0a8e673/pkg/cri/server/sandbox_stop.go#L189

Reply:
ok, I agree leaving network namespace is better.